### PR TITLE
fix:Signals working properly for testing test_job.py

### LINKED ISSUE #2842: SIGTERM handling not compatible with Windows
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/issues/2901

Originally created by @aaronsteers on 2021-08-18 22:02:56

---

As described in [#2743](https://gitlab.com/meltano/meltano/-/issues/2743), the SIGTERM handling is not currently supported on Windows.

Here is a possible explanation with workaround: https://stackoverflow.com/questions/26061814/signal-handling-in-windows

### PR DESCRIPTION
**Pull request by visch**
Closes #2842

### CODE REVIEW

File: tests/meltano/core/job/test_job.py [hunk 1]
```diff
@@ -176,3 +183,13 @@ def test_fail_stale(self):
...
+
+
+def windows_signal_process(signal: int):
+    # Replace with signal.raise_signal once Python 3.7 has been dropped
+    # https://stackoverflow.com/questions/35772001/how-to-handle-a-signal-sigint-on-a-windows-os-machine
+    import ctypes
+
+    ucrtbase = ctypes.CDLL("ucrtbase")
+    c_raise = ucrtbase["raise"]
+    c_raise(signal)
```
Comment (aaronsteers) [inline tests/meltano/core/job/test_job.py:195]: Overall, this looks really good. Thanks very much for creating this, @visch. 👍 

I like the comment and inline URL but what do you think of swapping this with a link to a new issue on Meltano side?

Perhaps something like "Remove/replace Windows 'raise' workaround in after Python 3.7 EOL."

And that issue can have the additional info and context?

File: tests/meltano/core/job/test_job.py [hunk 2]
```diff
@@ -89,10 +91,12 @@ async def test_run_failed(self, session):
...
     async def test_run_interrupted(self, session):
         subject = self.sample_job({"original_state": 1}).save(session)
-
         with pytest.raises(KeyboardInterrupt):
             async with subject.run(session):
-                psutil.Process().send_signal(signal.SIGINT)
+                if platform.system() == "Windows":
+                    windows_signal_process(signal.SIGINT)
+                else:
+                    psutil.Process().send_signal(signal.SIGINT)
```
Comment (aaronsteers) [inline tests/meltano/core/job/test_job.py:99]: What about moving this conditional check and the special handling into the helper function also?

```suggestion
                send_process_signal(signal.SIGINT)
```

And that function can own properly sending the signal depending on the OS check. What do you think?

Comment (visch) [reply tests/meltano/core/job/test_job.py]: Done, much cleaner!

File: tests/meltano/core/job/test_job.py (hunk 1 above)
Comment (visch) [reply tests/meltano/core/job/test_job.py]: Done, makes a lot of sense, left Python 3.7 in there so when someone greps for 3.7 later hopefully it'll come up

**Review by pandemicsyn — APPROVED** [review]
Don't have a windows system to manually test on anymore but looks pretty benign. lgtm.

**Review by edgarrmondragon — APPROVED** [review]

**Comment by visch** [comment]
Can we get this merged? @pandemicsyn

**Review by WillDaSilva — APPROVED** [review]

**Pull request merged**